### DISCUSSION TITLE
Permission anomaly?

### DISCUSSION THREAD

**Discussion by amoshalava** [vestrel00/contacts-android#171] [Q&A] [answered]
I am using kotlin but i can't figure out how to use your permissions with launch with my app from Android Studio. So I got some code that request the permission but then it acts strange. I answer YES to give permission but the code doesn't add any new contacts. If I kill the app. Go to app settings and permissions. Set contacts to deny. Then set back to allow. it works just fine. So when my program asks it set its permissions correctly but doesn't work. but when set the permission manually it works great. any suggestions? I get the same issue with TedPermission installed. Do you have a sample where you use the launch command with your permissions from an android kotlin project?

**Comment by vestrel00** [comment]
Hey! 

> I get the same issue with TedPermission installed.

I took a look at [TedPermission](https://github.com/ParkSangGwon/TedPermission), it looks good with first-class Kotlin Coroutine support! I was looking for a replacement for [Dexter](https://github.com/Karumi/Dexter) since it is archived and does not have first-class support for Coroutines. TedPermission looks like a great alternative!!! Thank you for letting me know ❤️ 🔥 I'll make note of this in #101 😁 

> Do you have a sample where you use the launch command with your permissions from an android kotlin project?

Yes! You can run the `sample` module; https://github.com/vestrel00/contacts-android/tree/main/sample . Please follow the instructions in the [setup guide](https://github.com/vestrel00/contacts-android/tree/main/sample) to get the sample app running locally on your machine.

<img width="905" alt="Screen Shot 2022-01-18 at 7 10 30 PM" src="https://user-images.githubusercontent.com/4332753/150068043-8ff10a68-da5d-4a8e-9cda-9aa3b6dd9c8b.png">

Please try it out and let me know if you are also able to reproduce the issue there.

> So I got some code that request the permission but then it acts strange. I answer YES to give permission but the code doesn't add any new contacts. If I kill the app. Go to app settings and permissions. Set contacts to deny. Then set back to allow. it works just fine. So when my program asks it set its permissions correctly but doesn't work. but when set the permission manually it works great.

Could you share the relevant parts of your code so I can see it?

**Reply by vestrel00** [reply]
Also, please tell me more information on the emulator or physical device you are using.

1. What is the API Level?
2. What is the model?

Also, are you able to reproduce this issue using a pixel emulator running API 31?

**Reply by amoshalava** [reply]
So using the TedPermission like this?
`            TedPermission.create()
                .setPermissionListener(permissionlistener)
                .setDeniedMessage("If you reject permission,you can not use this service\n\nPlease turn on permissions at [Setting] > [Permission]")
                .setPermissions(Manifest.permission.WRITE_CONTACTS)
                .check();
`

is that the only permission to ask for to delete and add contacts?

**Reply by vestrel00** [reply]
Inserting contacts (using this library) require all of these permissions;
- `android.permission.WRITE_CONTACTS`
- `android.permission.GET_ACCOUNTS`

Deleting requires;
- `android.permission.WRITE_CONTACTS`

For more info on insert and delete, see;

- https://github.com/vestrel00/contacts-android/blob/main/howto/howto-insert-contacts.md
- https://github.com/vestrel00/contacts-android/blob/main/howto/howto-delete-contacts.md

**Comment by vestrel00** [comment, marked as answer]
I will be replacing the current permissions library in #101. I'll be closing this for now since I didn't get much details about the anomaly.
